sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/ConnectionProvider.scala

```diff
@@ -22,12 +22,15 @@ import java.util.ServiceLoader
 
 import scala.collection.mutable
 
+import org.apache.spark.{SparkConf, SparkEnv}
 import org.apache.spark.internal.Logging
 import org.apache.spark.security.SecurityConfigurationLock
 import org.apache.spark.sql.jdbc.JdbcConnectionProvider
 import org.apache.spark.util.Utils
 
 private[jdbc] object ConnectionProvider extends Logging {
+  private val providerEnabledConfig = "spark.security.jdbc.connection.provider.%s.enabled"
```

Comment:
Why didn't you put this config in `SQLConfig`? We assume users don't care about it? I read the PR description and I thought first we would add a config like `spark.sql.sources.useV1SourceList`;
```
For example;


  val USE_JDBC_CONN_PROVIDER_LIST = buildConf("spark.sql.sources.useJdbcConnProviderList")
    .internal()
    .doc("...")
    .version("3.1.0")
    .stringConf
    .createWithDefault("db2,mysql,postgresql,...")
```

Reply:
This kind of pattern `provider.{providerName}.enabled` is used often in other non-SQL codebase e.g.) `"spark.security.credentials.%s.enabled"` at `HadoopDelegationTokenManager`. I think he's more used to this kind of pattern.

Since we're in SQL, maybe it's better to stick to the suggestion above if possible.

Reply:
Ah, I see. Thanks for the info. Adding it in `SQLConf` looks better if possible.

Reply:
It makes sense to make it SQL like so adapting. What I think however we need blacklist and not whitelist. Having a blacklist doesn't expect any configuration from user when new provider added.

Reply:
Just changed it.